### PR TITLE
improve pandas dataframe checking

### PR DESCRIPTION
**Pull request by gcamargo1**
Here's a suggestion to guarantee a positive pandas dataframe.

### CODE REVIEW

File: pycaret/internal/tabular.py [hunk 1]
```diff
@@ -283,12 +283,16 @@ def setup(
...
-        raise TypeError("data passed must be of type pandas.DataFrame")
+    if not isinstance(data, pd.DataFrame):
+        raise TypeError(f"data passed must be of type pandas.DataFrame")
+    if data.shape[0] == 0:
+        raise ValueError(f"data passed must be a positive dataframe")
 
     # checking train size parameter
     if type(train_size) is not float:
         raise TypeError("train_size parameter only accepts float value.")
+    if train_size <= 0:
```
Comment (Yard1) [inline pycaret/internal/tabular.py:294]: Would it also be possible to make sure it's not bigger than 1?

File: pycaret/internal/tabular.py [hunk 2]
```diff
@@ -300,7 +304,9 @@ def setup(
 
     # checking target parameter
     if not _is_unsupervised(ml_usecase) and target not in data.columns:
-        raise ValueError("Target parameter doesnt exist in the data provided.")
+        raise ValueError(
+            f"Target parameter: {target} does not exist in the data" f" provided."
```
Comment (Yard1) [inline pycaret/internal/tabular.py:308]: `data" f" provided."` to  `data provided."`

File: pycaret/internal/tabular.py [hunk 3]
```diff
@@ -332,23 +338,28 @@ def setup(
     # checking imputation type
     allowed_imputation_type = ["simple", "iterative"]
     if imputation_type not in allowed_imputation_type:
-        raise ValueError("imputation_type param only accepts 'simple' or 'iterative'")
+        raise ValueError(
+            "imputation_type parameter only accepts 'simple' or " "'iterative'."
```
Comment (Yard1) [inline pycaret/internal/tabular.py:342]: `or " "'iterative'."` to `or 'iterative'."`

File: pycaret/internal/tabular.py [hunk 4]
```diff
@@ -499,14 +509,23 @@ def setup(
 
     # check rare_level_threshold
     if type(rare_level_threshold) is not float:
-        raise TypeError("rare_level_threshold must be a float between 0 and 1.")
+        raise TypeError("rare_level_threshold parameter must be a float")
```
Comment (Yard1) [inline pycaret/internal/tabular.py:512]: This could be just one check (one if for float + range)

**Review by Yard1 — CHANGES_REQUESTED** [review]
Thanks for your contribution. I just have a few nitpicks.

**Comment by gcamargo1** [comment]
Hi @Yard1 , tried to address all of your comments, please let me know if it makes sense to you.

**Comment by Yard1** [comment]
@gcamargo1 Thanks for your contribution.

**Comment by gcamargo1** [comment]
@Yard1 , I use black, which I'm not sure if it deals in the docstring.

**Pull request merged**